src/gluonts/transform/field.py

```diff
@@ -127,4 +127,4 @@ def __init__(self, input_fields: List[str]) -> None:
         self.input_fields = input_fields
 
     def map_transform(self, data: DataEntry, is_train: bool) -> DataEntry:
-        return {f: data[f] for f in self.input_fields}
+        return {f: data[f] for f in self.input_fields if f in data.keys()}
```

Comment:
maybe this results in silent errors in other cases (ignoring required fields simply because they are not in the data)?

Reply:
Good point. I don't think the error would be silent in this case (at least mxnet models will complain for sure!). This change is required to have optional fields in the network forward passes, though. I think the fundamental question that we have to ask: 

Do we want to have optional inputs for the networks or not?